python/tests/test_topology.py

```diff
@@ -4647,16 +4665,17 @@ def example_intervals(self, tables):
 
     def do_keep_intervals(
             self, tables, intervals, simplify=True, record_provenance=True):
-        t1 = simple_keep_intervals(tables, intervals, simplify, record_provenance)
-        t2 = tables.keep_intervals(intervals, simplify, record_provenance)
-        t3 = t2.copy()
+        t1 = tables.copy()
+        simple_keep_intervals(t1, intervals, simplify, record_provenance)
+        t2 = tables.copy()
+        t2.keep_intervals(intervals, simplify, record_provenance)
         self.assertEqual(len(t1.provenances), len(t2.provenances))
```

Comment:
Might as well use the new tables_equal function here.

Reply:
Good spot, thanks. There are probably other places where this can be done too, but I'll get round to that in another PR